pulp_container/app/registry_api.py

```diff
@@ -1009,3 +1049,61 @@ def get_response_data(signatures):
             }
             data.append(signature)
         return {"signatures": data}
+
+    def put(self, request, path, pk):
+        """Create a new signature from the received data."""
+        _, repository = self.get_dr_push(request, path)
+
+        try:
+            manifest = models.Manifest.objects.get(digest=pk)
+        except models.Manifest.DoesNotExist:
+            raise ManifestNotFound(reference=pk)
+
+        signature_payload = request.META["wsgi.input"].read(SIGNATURE_PAYLOAD_MAX_SIZE)
+        signature_raw = base64.b64decode(json.loads(signature_payload)["content"])
```

Comment:
what if the signature's payload will exceed the limit, will json be able to load and access the `content`?

Reply:
we need to assert that body of PUT request is correct, namely that each signature instance contains schemaversion ==2 to and type == atomic. Name also should be required and i think it has some rules